File: 프로그래머스/unrated/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.java
class Solution {
    public long solution(int r1, int r2) {
        long answer = 0;
        
        long r1_pow = (long)r1 * r1;
        long r2_pow = (long)r2 * r2;
        for(long x = 1; x <= r2; x++) {
            long x_pow = x * x;
            long max_y = (long)Math.sqrt(r2_pow - x_pow);
            long min_y = 0;
            if(x < r1)
                min_y = (long)Math.ceil(Math.sqrt(r1_pow - x_pow));
            answer += max_y - min_y + 1;
        }
        
        return answer << 2;
    }
}

File: 프로그래머스/unrated/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.py
from math import sqrt, ceil

def solution(r1, r2):
    answer = 0
    
    r1_pow = r1 * r1
    r2_pow = r2 * r2
    for x in range(1, r2 + 1):
        x_pow = x * x
        max_y = int(sqrt(r2_pow - x_pow))
        min_y = ceil(sqrt(r1_pow - x_pow)) if x < r1 else 0
        answer += max_y - min_y + 1
    
    return answer << 2